### DISCUSSION TITLE
GPL exception clarification

### DISCUSSION THREAD

**Discussion by GeorgePriestner-Eaton** [bacnet-stack/bacnet-stack#582] [General]
Hi @skarg,

I had a question about the GPL license the bacnet-stack is registered under; the exception states:

_This BACnet protocol stack implementation is specifically designed for the embedded BACnet appliance, using a GPL with exception license (like eCos), which means that any changes to the core code that are distributed are shared, but the BACnet library **can be linked** to proprietary code without the proprietary code becoming GPL._

Does it matter which method of linking is used? I.e. we are looking to use in Zephyr (RTOS) and so would be statically linked (not dynamically linked) to our proprietary code.

Thanks!

cc @gregshue

**Comment by skarg** [comment]
The method of linking does not matter for GPL-2.0-or-later WITH [GCC-exception-2.0](https://spdx.org/licenses/GCC-exception-2.0.html) license.

Note:  GPL-2.0-or-later WITH GCC-exception-2.0 license only applies to the core BACnet C modules used for encoding/decoding/state-machine-operation.  Every file includes a copyright and their specific license in the file header.

See also a task to convert all the text license info to SPDX format: #55

**Reply by GeorgePriestner-Eaton** [reply]
Thank you for clarification on that @skarg. It's clear to me now.

George
